jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -587,19 +607,22 @@ public void removeListener(Listener listener)
     public void doStart() throws Exception
     {
         if (LOG.isDebugEnabled())
-            LOG.debug("Scanner start: rprtExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
-                      _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
+            LOG.debug("Scanner start: scanInStart={}, reportExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
+                      _scanInStart, _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
 
-        if (_reportExisting)
+        if (_scanInStart)
```

Comment:
Could be something like:
``` java
private boolean _startupCalled = false;
 
public void doStart()
{
  if (!_deferredStart)
  {
    startup();
  }

  super.doStart();
}

public void doStop()
{
  _startupCalled = false;
}


public void startup()
{
   if (_startupCalled)
       return;
    _startupCalled = true;

   if (_reportExisting) 
   {
     // if files exist at startup, report them
     scan();
     scan(); // scan twice so files reported as stable
   }
   else
   {
      //just register the list of existing files and only report changes
      _prevScan = scanFiles();
   }
}
```

Then in the `ScanningAppProvider`'s `Server` `LifeCycle.Listener.lifeCycleStarted()` method it can call:
```java
   scanner.startup();
```

Reply:
Ah this is better than my suggestion, as:
 + reportExisting is still meaningful will a deferred start
 + there is no race with the scheduler (oh we should also do a call to schedule() in the startup method)


Not a huge fan of the "deferStartup" name as it is strange that `doStart` defers "startup".    Perhaps "deferScan" and do the check to call startup in the first call to scan()?

Reply:
Doh! Typo - missed out the call to `schedule()` in my suggested `startup()` method.

Reply:
Done.

Needed to start the scheduler in the right place as well.